### PR TITLE
Add rules for project detection

### PR DESCRIPTION
**Pull request by tmattio**
Currently, the extension is activated only when the user opens a supported file. This PR adds rules so that the extension is activated when opening an OCaml project.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,24 @@
     "onLanguage:dune-workspace",
     "onCommand:ocaml.select-sandbox",
     "onCommand:ocaml.open-terminal",
-    "onCommand:ocaml.open-terminal-select"
+    "onCommand:ocaml.open-terminal-select",
+    "onCommand:ocaml.open-terminal-select",
```
Comment (smorimoto) [inline package.json:35]: This seems to be an unintentional duplication.

Comment (tmattio) [reply package.json:35]: Indeed, thanks!

File: package.json [hunk 2]
```diff
@@ -31,7 +31,23 @@
...
     "onCommand:ocaml.select-sandbox",
     "onCommand:ocaml.open-terminal",
-    "onCommand:ocaml.open-terminal-select"
+    "onCommand:ocaml.open-terminal-select",
+    "workspaceContains:dune-project",
+    "workspaceContains:**/dune-project",
+    "workspaceContains:*.opam",
+    "workspaceContains:**/*.opam",
+    "workspaceContains:*.ml",
+    "workspaceContains:**/*.ml",
```
Comment (rgrinberg) [inline package.json:40]: Are you sure we need both `*.foo` and `**/*.foo`?

From the docs:

> ** to match any number of path segments, including none

So I think `**/.foo` should be enough.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 # Unreleased
 
 - Highlight token aliases in Menhir associativity declarations (#473)
+- The extension is not activated when an OCaml project is detected (#482)
```
Comment (rgrinberg) [inline CHANGELOG.md:6]: If we're describing a bug fix, we should allude to that with something like

> Fix activation of extension. The extension was not activated when an OCaml project was detected.

Or we can simply describe the new behavior:

> Activate the extension when workspace contains OCam, Reason sources or project marker files.

Comment (tmattio) [reply CHANGELOG.md:6]: Meant to write "now":
"The extension is now activated when an OCaml project is detected"

Comment (tmattio) [reply CHANGELOG.md:6]: But your version is more precise, will replace it :)

File: package.json (hunk 2 above)
Comment (tmattio) [reply package.json:37]: Seems to work as well.

**Review by rgrinberg — APPROVED** [review]

**Pull request merged**